### PR TITLE
feat: reenable check for visibility modifiers

### LINKED ISSUE #278: Disallow private fields without leading _
Time to fix this old TODO: https://github.com/angular/ts2dart/blob/6f3bbd896e7e712049eb4f2403e67f59932cf697/lib/declaration.ts#L426

### PR DESCRIPTION
**Pull request by harryterkelsen**
Fixes #278

### CODE REVIEW

File: lib/declaration.ts [hunk 1]
```diff
@@ -422,13 +426,11 @@ class DeclarationTranspiler extends base.TranspilerBase {
...
     this.visitEachIfPresent(decl.modifiers);
 
-    // Temporarily deactivated to make migration of Angular code base easier.
-    return;
-
     if (this.hasFlag(decl.modifiers, ts.NodeFlags.Protected)) {
       this.reportError(decl, 'protected declarations are unsupported');
       return;
     }
+    if (!decl.name) return;
```
Comment (mprobst) [inline lib/declaration.ts]: Could you add a comment explaining when this happens?

Comment (hterkelsen) [inline lib/declaration.ts]: I don't remember what was causing it when I put the check in, but I took it out again and the tests all pass, so I'll just leave it out

Comment (hterkelsen) [inline lib/declaration.ts]: Nevermind. It happens with constructors. Added comment

**Comment by mprobst** [comment]
LGTM, please merge.

**Comment by hterkelsen** [comment]
Should I merge even if there are several hundred places in Angular that will become errors if this check is turned on?

**Comment by mprobst** [comment]
Yes, but that makes you responsible to fix those in Angular when you upgrade ts2dart :-)

**Comment by hterkelsen** [comment]
I've already fixed all of the private fields without leading underscores. I would actually like to allow public fields with leading underscores since that is helpful for dart2js. What do you think?

**Comment by mprobst** [comment]
Merged as e288c35.

As discussed offline, let's make all those public fields leading with an underscore in angular `private`, as they should be.

**Pull request closed without merging**